bpfd/src/multiprog/xdp.rs

```diff
@@ -43,19 +43,20 @@ impl XdpDispatcher {
         mode: XdpMode,
         if_index: &u32,
         if_name: String,
-        programs: &mut [(Uuid, Program)],
+        programs: &mut [(&Uuid, &mut Program)],
```

Comment:
Yeah so if you wanted to take this all the way, this would become
```rust
programs: impl Iterator<Item=(&Uuid, &mut Program)>
```

Any calls to `.len()` get replaced with `.count()`
Any calls to `.first()` become `.peek()`
And then we're good 😉 

We might also - which I know you'll love - be able to remove the call to `sort` on the collected Vec.
It shouldn't matter if we attach the extensions out-of-order 😆 

Anyway, this is definitely new PR territory...

Reply:
I like It! leaving these comments open for the next PR